File: Questão10.java
abstract class Forma {
    public abstract double area();
}

class Retangulo extends Forma {
    private double largura;
    private double altura;

    public Retangulo(double largura, double altura) {
        this.largura = largura;
        this.altura = altura;
    }

    @Override
    public double area() {
        return largura * altura;
    }
}

class Circulo extends Forma {
    private double raio;

    public Circulo(double raio) {
        this.raio = raio;
    }

    @Override
    public double area() {
        return Math.PI * raio * raio;
    }
}

public class Main {
    public static void main(String[] args) {
        Forma[] formas = { new Retangulo(5, 10), new Circulo(7) };
        for (Forma forma : formas) {
            System.out.println("Área: " + forma.area());
        }
    }
}

File: Questão10.go
package main

import (
    "fmt"
    "math"
)


type Forma interface {
    Area() float64
}

type Retangulo struct {
    Largura, Altura float64
}


func (r Retangulo) Area() float64 {
    return r.Largura * r.Altura
}

type Circulo struct {
    Raio float64
}


func (c Circulo) Area() float64 {
    return math.Pi * c.Raio * c.Raio
}

func main() {
    formas := []Forma{Retangulo{5, 10}, Circulo{7}}
    for _, forma := range formas {
        fmt.Printf("Área: %.2f\n", forma.Area())
    }
}
